### PR TITLE
Rh pundit authorization

### PR DESCRIPTION
**Pull request by reherr**

### CODE REVIEW

File: app/policies/user_policy.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  def update?
+    edit?
+  end
+
+  def destroy?
+    edit?
+  end
+
+  def show?
```
Comment (ajmompr) [inline app/policies/user_policy.rb:25]: almost! Is your app displaying properly for users who are private. Try moving this logic to a method called show_photos? and setting show to true. Call show_photos? policy in your users#show template

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```
Comment (ajmompr) [inline app/views/users/show.html.erb]: define a new show_photos? policy method and call it here

File: how?
Comment (ajmompr) [inline how?:1]: what is this??

**Review by ajmompr — CHANGES_REQUESTED** [review]
Fantastic job; small tweaks to your user policy

File: app/policies/user_policy.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  def following?
+    show?
+  end
+
+  def followers?
+    show?
+  end
+
+  def show_photos?
+    user == current_user || !user.private? || user.followers.include?(current_user) || current_user.followers.include?(user)
```
Comment (ajmompr) [inline app/policies/user_policy.rb:46]: perf

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show_photos? %>
```
Comment (ajmompr) [inline app/views/users/show.html.erb:7]: perf

**Review by ajmompr — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,4 +1,9 @@
 class ApplicationController < ActionController::Base
+  include Pundit
+
+  after_action :verify_authorized, unless: :devise_controller?
+  after_action :verify_policy_scoped, only: :index, unless: :devise_controller?
```
Comment (borvux) [inline app/controllers/application_controller.rb:5]: good job adding for devise

File: app/policies/user_policy.rb (hunk 3 above)
Comment (borvux) [inline app/policies/user_policy.rb:46]: good job doing show_photos to differentiate between show

**Review by borvux — APPROVED** [review]
everything looks good to me!

**Pull request merged**